alica_engine/src/engine/RunningPlan.cpp

```diff
@@ -395,12 +395,19 @@ void RunningPlan::clearChildren()
  */
 void RunningPlan::adaptAssignment(const RunningPlan& replacement)
 {
+    double oldUtility = _assignment.getLastUtilityValue();
     _assignment.adaptTaskChangesFrom(replacement.getAssignment());
     const State* newState = _assignment.getStateOfAgent(getOwnID());
 
     bool reactivate = false;
 
     if (_activeTriple.state != newState) {
+        if (_ae->getTraceFactory() && _basicPlan && _basicPlan->getTraceContext().has_value()) {
+            _basicPlan->getTrace()->setLog({"TaskAssignmentChange", replacement.getActiveEntryPoint()->getName() + std::to_string(replacement.getActiveEntryPoint()->getId()) + "\n"
```

Comment:
this won't work because the tracing in _basicPlan is not mutex protected & is meant to be only called from the scheduler thread i.e. from basic plan/behaviour class. I think we should explore moving the tracing to the running plan instead & then the basic plan/behaviour can use the traces from there because the running plan object is guaranteed to be alive as long as the basic behaviour/plan is alive.